web-bundle/src/main/java/com/graphhopper/replica/StreetEdgeExporter.java

```diff
@@ -28,7 +28,9 @@ public class StreetEdgeExporter {
     private static final Map<String, String> ACCESSIBILITY_MODE_MAP = Map.of(
             "car", "CAR",
             "bike", "BIKE",
-            "foot", "PEDESTRIAN"
+            "foot", "PEDESTRIAN",
+            "truck", "TRUCK",
+            "small_truck", "SMALL_TRUCK"
     );
```

Comment:
you mentioned in the pr desc how we'll likely generate a `(OSM Way ID, mode, direction) -> (boolean representing accessibility)` mapping from the street export to facilitate upstream validations, and that this assumes the access for all the stable edge ids for a way is identical. should we start validating that explicitly when producing the street export?

Reply:
Yes, I think that's a good idea!

I took a crack at this in [41ced0c](https://github.com/replicahq/graphhopper/pull/171/commits/41ced0cd307c42c203cde8095e7ce2571653b49b); it works for the street export unit test (which creates a small export from a cutout of MNC), but we should definitely test it in the nationwide street exporter too, to make sure it doesn't use up too much memory.

In order to add this validation, we basically needed to store something equivalent to the mapping you described here between OSM Way/mode/direction and accessibility, so that's an added plus. If this approach looks good, we could consider a follow-up that simply outputs this mapping as a file, and that's what could be used in a future GUI to enforce directional accessibility constraints are enforced. (also, we wouldn't need to rebuild the TN CDP at all to incorporate this, as the street export now lives in the road network cdp)